starlark/prog_avail_local_test.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package starlark
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"go.starlark.net/starlark"
+)
+
+func TestProgAvailLocalScript(t *testing.T) {
+	tests := []struct {
+		name   string
+		prog   string
+		exists bool
+	}{
+		{
+			name:   "prog_avail_local checks for 'go'",
+			prog:   "go",
+			exists: true,
+		},
+		{
+			name:   "prog_avail_local checks for 'nonexistant'",
+			prog:   "nonexistant",
+			exists: false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			script := fmt.Sprintf(`path=prog_avail_local(prog="%v")`, test.prog)
+			exe := New()
+			if err := exe.Exec("test.star", strings.NewReader(script)); err != nil {
+				t.Fatal(err)
+			}
+
+			resultVal := exe.result["path"]
+			if resultVal == nil {
+				t.Fatal("prog_avail_local() should be assigned to a variable for test")
+			}
+
+			result, ok := resultVal.(starlark.String)
+			if !ok {
+				t.Fatal("prog_avail_local() should return a string")
+			}
+
+			if (len(string(result)) == 0) == test.exists {
+				if test.exists {
+					t.Fatalf("expecting '%v' to exists but 'prog_avail_local' didnt find it", test.prog)
+				} else {
+					t.Fatalf("expecting '%v' to not exists but 'prog_avail_local' found it", test.prog)
+
+				}
+			}
+		})
+	}
+}
```

Comment:
Add tests to run the code from starlark script as is done here for instance

https://github.com/vmware-tanzu/crash-diagnostics/blob/8b79e7dc6ecf1dacbfa1270eb7f7946409269a8c/starlark/run_local_test.go#L45

Reply:
The test is running and testing the using a starlark script. In [this line](https://github.com/ykakarap/crash-diagnostics/blob/prog_avail/starlark/prog_avail_local_test.go#L34) you can see that it is creating a simple script to test the `prog_avail_local` function.

Reply:
@ykakarap oh I see now.  You should consider adding the script in the test struct.  That way you can test different scenario such as 
```go
		{
			name:   "prog_avail_local checks for 'go'",
			prog:   "go",
			exists: true,
                      script: "prog_avail_local(prog="go")
		},
		{
			name:   "prog_avail_local default arg",
			prog:   "go",
			exists: true,
                      script: "prog_avail_local("go")
		},
		{
			name:   "prog_avail_local full path",
			prog:   "go",
			exists: true,
                      script: "prog_avail_local("/full/path/to/go")
		},
```

Reply:
@vladimirvivien I have updated tests